### PR TITLE
Improvement:  Filterable Project Select for Audit Logs

### PR DESCRIPTION
**Pull request by scott-ray-wilson**
# Description 📣

This PR updates the project select to be filterable on audit logs. Also includes minor UI adjustments

CleanShot 2024-11-26 at 15 11 03@2x

## Type ✨

- [ ] Bug fix
- [ ] New feature
- [x] Improvement
- [ ] Breaking change
- [ ] Documentation

# Tests 🛠️

```sh
# Here's some code block to paste some code snippets
```

---

- [x] I have read the [contributing guide](https://infisical.com/docs/contributing/getting-started/overview), agreed and acknowledged the [code of conduct](https://infisical.com/docs/contributing/getting-started/code-of-conduct). 📝

### CODE REVIEW

**Review by DanielHougaard — CHANGES_REQUESTED** [review]
left one comment

File: frontend/src/components/v2/DatePicker/DatePicker.tsx [hunk 1]
```diff
@@ -14,6 +14,7 @@ export type DatePickerProps = Omit<DayPickerProps, "selected"> & {
   onChange: (date?: Date) => void;
   popUpProps: PopoverProps;
   popUpContentProps: PopoverContentProps;
+  dateFormat?: string;
```
[the remark this answers is not in the document]
Comment (scott-ray-wilson) [reply frontend/src/components/v2/DatePicker/DatePicker.tsx:17]: I wish `date-fns` format was strongly typed but it just takes a string - given this is a 1:1 param I opted to keep it the same. I can do the two we're using but then anytime anyone wants to use another format they'll need to add it. `date-fns` throws an error on invalid formats so not like anything will slip through

[the remark this answers is not in the document]
Comment (DanielHougaard) [reply frontend/src/components/v2/DatePicker/DatePicker.tsx:17]: Hmm I think I would prefer to have it typed like in the comment I wrote above, but I'll leave it up to you!

**Review by DanielHougaard — APPROVED** [review]

File: frontend/src/components/v2/DatePicker/DatePicker.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (scott-ray-wilson) [reply frontend/src/components/v2/DatePicker/DatePicker.tsx]: addressed!

**Pull request merged**